packages/protocol/contracts/stability/Exchange.sol

```diff
@@ -297,6 +373,27 @@ contract Exchange is
     return numerator.unwrap().div(denominator.unwrap());
   }
 
+  /**
+   * @dev Returns the amount of sellToken a user would need to exchange to receive buyAmount of
+   * buyToken.
+   * @param buyAmount The amount of buyToken the user would like to purchase.
+   * @param sellGold `true` if gold is the sell token
+   * @return The corresponding sellToken amount.
+   */
+  function _getSellTokenAmount(uint256 buyAmount, bool sellGold) private view returns (uint256) {
```

Comment:
This is a copy of `getSellTokenAmount`, I copied just because that is what was done for `_getBuyTokenAmount`/`getBuyTokenAmount`. Is there a reason for copying rather than having `getBuyTokenAmount` call `_getBuyTokenAmount`?

Reply:
I would like to get more context on this too. I did noticed that `_getBuyTokenAmount` is calling `_getBuyAndSellBuckets` instead of `getBuyAndSellBuckets` as in your function below.

I think the logic there is that `_getBuyAndSellBuckets` doesn't try to update the buckets because that is already done in `sell`/`buy` with the `updateBucketsIfNecessary` modifier.

Reply:
I also have this question.

Reply:
+1 and maybe @m-chrzan has some more background on this?

Reply:
Ah I didn't realize that @nvtaveras, that makes sense to me. I'll change right now to use `_getBuyAndSellBuckets`. Thanks!

Context from familiar folks would still be helpful

Reply:
Yeah @nvtaveras is right, but I think we can do better here. Wdyt of the following example? We would do the same for the `buy` versions of these functions as well
```
function getSellTokenAmount(uint256 buyAmount, bool sellGold) external view returns (uint256) {
    (uint256 buyTokenBucket, uint256 sellTokenBucket) = getBuyAndSellBuckets(sellGold);
    return _getSellTokenAmount(buyTokenBucket, sellTokenBucket, buyAmount);
}

function _getSellTokenAmount(uint256 buyTokenBucket, uint256 sellTokeBucket, buyAmount) external view returns (uint256) {
  // Shared code in the current versions of _getSellTokenAmount and getSellTokenAmount goes here
}

function buy(uint256 buyAmount, uint256 maxSellAmount, bool buyGold) {
  (uint256 buyTokenBucket, uint256 sellTokenBucket) = _getBuyAndSellBuckets(!buyGold);
   uint256 sellAmount = _getSellTokenAmount(buyTokenBucket, sellTokenBucket, buyAmount);
   ...
}
```